plugin/src/main/java/org/opensearch/ml/jobs/MLBatchTaskUpdateJobRunner.java

```diff
@@ -0,0 +1,159 @@
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.jobscheduler.spi.ScheduledJobRunner;
+import org.opensearch.jobscheduler.spi.utils.LockService;
+import org.opensearch.ml.common.transport.task.MLTaskGetAction;
+import org.opensearch.ml.common.transport.task.MLTaskGetRequest;
+import org.opensearch.ml.task.MLTaskManager;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.threadpool.ThreadPool;
+
+public class MLBatchTaskUpdateJobRunner implements ScheduledJobRunner {
+    private static final Logger log = LogManager.getLogger(ScheduledJobRunner.class);
+
+    private static MLBatchTaskUpdateJobRunner INSTANCE;
+
+    public static MLBatchTaskUpdateJobRunner getJobRunnerInstance() {
+        if (INSTANCE != null) {
+            return INSTANCE;
+        }
+        synchronized (MLBatchTaskUpdateJobRunner.class) {
+            if (INSTANCE != null) {
+                return INSTANCE;
+            }
+            INSTANCE = new MLBatchTaskUpdateJobRunner();
+            return INSTANCE;
+        }
+    }
+
+    private ClusterService clusterService;
+    private ThreadPool threadPool;
+    private Client client;
+    private MLTaskManager taskManager;
+    private boolean initialized;
+
+    private MLBatchTaskUpdateJobRunner() {
+        // Singleton class, use getJobRunner method instead of constructor
+    }
+
+    public void setClusterService(ClusterService clusterService) {
+        this.clusterService = clusterService;
+    }
+
+    public void setThreadPool(ThreadPool threadPool) {
+        this.threadPool = threadPool;
+    }
+
+    public void setClient(Client client) {
+        this.client = client;
+    }
+
+    public void initialize(
+        final ClusterService clusterService,
+        final ThreadPool threadPool,
+        final Client client,
+        final MLTaskManager taskManager
+    ) {
+        this.clusterService = clusterService;
+        this.threadPool = threadPool;
+        this.client = client;
+        this.taskManager = taskManager;
+        this.initialized = true;
+    }
+
+    @Override
+    public void runJob(ScheduledJobParameter scheduledJobParameter, JobExecutionContext jobExecutionContext) {
+        if (initialized == false) {
+            throw new AssertionError("this instance is not initialized");
+        }
+
+        final LockService lockService = jobExecutionContext.getLockService();
+
+        lockService.acquireLock(scheduledJobParameter, jobExecutionContext, ActionListener.wrap(lock -> {
+            if (lock == null) {
+                return;
+            }
+
+            String jobName = scheduledJobParameter.getName();
+            log.info("Starting job execution for job ID: {} at {}", jobName, Instant.now());
+
+            if (taskManager == null) {
+                log.error("TaskManager not initialized. Cannot run batch task polling job");
+                return;
+            }
```

Comment:
This is a bit strange because taskManager is never used in this function runJob(). If this is needed in the GetTask Request, it will be checked in that action anyway.

Reply:
Thanks for the catch. Yeah actually we don't need it. Initially the logic was implemented differently which required taskManager. It is safe to be removed now.